libraries/shared/storage/backfill/storage_value_loader.go

```diff
@@ -33,32 +34,35 @@ type StorageValueLoader struct {
 	HeaderRepo      datastore.HeaderRepository
 	StorageDiffRepo storage2.DiffRepository
 	initializers    []storage.TransformerInitializer
-	blockNumber     int64
+	startingBlock   int64
+	endingBlock     int64
 }
 
 func (r *StorageValueLoader) Run() error {
 	addressToKeys, getKeysErr := r.getStorageKeys()
 	if getKeysErr != nil {
 		return getKeysErr
 	}
-	header, getHeaderErr := r.HeaderRepo.GetHeader(r.blockNumber)
-	if getHeaderErr != nil {
-		return getHeaderErr
+	headers, getHeadersErr := r.HeaderRepo.GetHeadersInRange(r.startingBlock, r.endingBlock)
+	if getHeadersErr != nil {
+		return getHeadersErr
 	}
 
-	for address, keys := range addressToKeys {
-		persistStorageErr := r.getAndPersistStorageValues(address, keys, r.blockNumber, header.Hash)
-		if persistStorageErr != nil {
-			return persistStorageErr
+	for _, header := range headers {
```

Comment:
This solution seems a little clunky, since it'll persist a new "diff" for each block even if nothing changed in that block. I couldn't really find a way around this behavior via Geth's API, but it seems like something we should be able to handle in the repository layer, or even in the DB itelf.

Reply:
Yeah I'm fine with this - maybe worth adding a story for pruning duplicate diffs? I think we have one for pruning obsolete diffs but that refers to unwatched contracts iirc

Reply:
https://makerdao.atlassian.net/browse/VDB-1261